weave/trace_server/opentelemetry/attributes.py

```diff
@@ -313,199 +308,90 @@ def unflatten_key_values(
+                result[key] = value
+                break
+    return result
 
-class OpenTelemetryAttributes(Attributes):
-    def get_weave_usage(self) -> LLMUsageSchema:
-        prompt_tokens = self.get_attribute_value(
-            ot.SpanAttributes.LLM_USAGE_PROMPT_TOKENS
-        )
-        completion_tokens = self.get_attribute_value(
-            ot.SpanAttributes.LLM_USAGE_COMPLETION_TOKENS
-        )
-        total_tokens = self.get_attribute_value(
-            ot.SpanAttributes.LLM_USAGE_TOTAL_TOKENS
-        )
-        return LLMUsageSchema(
-            prompt_tokens=int(prompt_tokens) if prompt_tokens else None,
-            completion_tokens=int(completion_tokens) if completion_tokens else None,
-            total_tokens=int(total_tokens) if total_tokens else None,
-        )
 
-    def get_weave_summary(self) -> SummaryInsertMap:
-        return SummaryInsertMap(usage={"usage": self.get_weave_usage()})
+def get_weave_attributes(attributes: dict[str, Any]) -> dict[str, Any]:
+    value = parse_weave_values(attributes, ATTRIBUTE_KEYS)
+    return value
 
-    def get_weave_outputs(self) -> Any:
-        outputs: dict[str, Any] = (
-            self.get_attribute_value(ot.SpanAttributes.LLM_COMPLETIONS) or {}
-        )
-        return to_json_serializable(outputs)
 
-    def get_weave_inputs(self) -> Any:
-        inputs: dict[str, Any] = (
-            self.get_attribute_value(ot.SpanAttributes.LLM_PROMPTS) or {}
-        )
-        return to_json_serializable(inputs)
-
-    def get_weave_attributes(
-        self, extra: Optional[dict[str, Any]] = None
-    ) -> dict[str, Any]:
-        max_tokens = self.get_attribute_value(ot.SpanAttributes.LLM_REQUEST_MAX_TOKENS)
-        system = self.get_attribute_value(ot.SpanAttributes.LLM_SYSTEM)
-        kind = self.get_attribute_value(ot.SpanAttributes.TRACELOOP_SPAN_KIND)
-        model = self.get_attribute_value(ot.SpanAttributes.LLM_RESPONSE_MODEL)
-
-        attributes = {
-            "system": str(system) if system else None,
-            "max_tokens": int(max_tokens) if max_tokens else None,
-            "kind": str(kind) if kind else None,
-            "model": str(model) if model else None,
-        }
+def get_weave_usage(attributes: dict[str, Any]) -> dict[str, Any]:
+    usage = parse_weave_values(attributes, USAGE_KEYS)
+    if (
+        "prompt_tokens" in usage
+        and "completion_tokens" in usage
+        and "total_tokens" not in usage
+    ):
+        usage["total_tokens"] = usage["prompt_tokens"] + usage["completion_tokens"]
+    if (
+        "input_tokens" in usage
+        and "output_tokens" in usage
+        and "total_tokens" not in usage
+    ):
+        usage["total_tokens"] = usage["prompt_tokens"] + usage["completion_tokens"]
+    return usage
 
-        if extra:
-            attributes.update(extra)
 
-        return to_json_serializable(attributes)
+def get_weave_inputs(_: list[SpanEvent], attributes: dict[str, Any]) -> dict[str, Any]:
```

Comment:
Passing the events here even though they are unused because there are a few exporters which make use of event attributes. We don't support this yet and should add it as a followup in the next release.